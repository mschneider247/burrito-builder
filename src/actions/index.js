export const setOrders = orders => ({
  type: 'SET_ORDERS',
  orders
});

export const placeOrder = order => ({
  type: 'PLACE_ORDER',
  order
})
