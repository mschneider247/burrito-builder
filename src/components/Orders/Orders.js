import React from 'react';
import { Component } from 'react';
import './Orders.css';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setOrders, removeOrder } from '../../actions';
import { getOrders } from '../../apiCalls'

export class Orders extends Component {

  componentDidMount() {
    getOrders()
      .then(data => this.props.setOrders(data.orders))
      .catch(err => console.error('Error fetching:', err));
  }

  removeOrder = (index) => {
    this.props.removeOrder(index)
    // deleteOrder(??)
  }

  orderEls = () => this.props.orders.map((order, index) => {
    return (
      <div className="order" key={index}>
        <h3>{order.name}</h3>
        <ul className="ingredient-list">
          {order.ingredients.map((ingredient, index) => {
            return <li key={index}>{ingredient}</li>
          })}
        </ul>
        <button onClick={() => this.removeOrder(index)}>Order Finished! DELETE</button>
      </div>
    )
  });

  render() {
    let orders = this.orderEls()
    let ordersLength = orders.length;
    return (
      <section>
        { ordersLength > 0 ? orders : <p>No orders yet!</p> }
      </section>
    )
  }
}

export const mapStateToProps = ({ orders }) => ({
  orders
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setOrders,
    removeOrder,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(Orders);