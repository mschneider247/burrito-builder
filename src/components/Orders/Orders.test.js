import React from 'react';
import { shallow } from 'enzyme';
import { Orders, mapDispatchToProps, mapStateToProps } from './Orders'
import { setOrders } from '../../actions';

describe('Orders container', () => {
  const wrapper = shallow(<Orders orders={[]}/>);
  
  it('Should match its snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('Should return order Elements when orderEls is called', () => {
    const expectedOrders = []
    const orders = wrapper.instance().orderEls([])
    expect(orders).toEqual(expectedOrders)
  })
});

describe('Orders mapDispatchToProps', () => {

  it('Should dispatch orders when setOrders is called', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setOrders([{name: 'steve', ingredients: 'food'}]);
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setOrders([{name: 'steve', ingredients: 'food'}]);
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});

describe('Orders mapStateToProps', () => {

  it('Should have orders set in the redux store', () => {
    const mockState = { orders: [{}, {}, {}] };
    const expected = { orders: [{}, {}, {}] };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  })
})