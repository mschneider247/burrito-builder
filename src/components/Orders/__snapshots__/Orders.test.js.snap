// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders container Should match its snapshot 1`] = `
<section>
  <p>
    No orders yet!
  </p>
</section>
`;
